### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by tyiu**

### CODE REVIEW

File: tempest/src/test/kotlin/app/cash/tempest/DynamoDbViewTest.kt [hunk 1]
```diff
@@ -107,13 +108,13 @@ class DynamoDbViewTest {
...
 
     // This fails because playlist_size is already 1.
-    assertThatExceptionOfType(ConditionalCheckFailedException::class.java)
-      .isThrownBy {
-        musicTable.playlistInfo.save(
-          playlistInfoV2,
-          ifPlaylistVersionIs(playlistInfoV1.playlist_version)
-        )
-      }
+    val exception = assertThrows<AmazonServiceException> {
```
Comment (tyiu) [inline tempest/src/test/kotlin/app/cash/tempest/DynamoDbViewTest.kt:111]: Unfortunate consequence of a bug in the DynamoDb Local client where it throws a different exception than what would be thrown in a real-world scenario.

File: tempest/src/test/kotlin/app/cash/tempest/LogicalDbTransactionTest.kt [hunk 2]
```diff
@@ -190,9 +191,10 @@ class LogicalDbTransactionTest {
     // Introduce a race condition.
     musicTable.playlistInfo.save(playlistInfoV2)
 
-    assertFailsWith<TransactionCanceledException> {
+    val exception = assertThrows<AmazonServiceException> {
```
Comment (tyiu) [inline tempest/src/test/kotlin/app/cash/tempest/LogicalDbTransactionTest.kt:194]: Unfortunate consequence of a bug in the DynamoDb Local client where it throws a different exception than what would be thrown in a real-world scenario.

File: tempest/src/test/kotlin/app/cash/tempest/DynamoDbViewTest.kt [hunk 3]
```diff
@@ -75,10 +76,10 @@ class DynamoDbViewTest {
     musicTable.albumInfo.save(albumInfo, ifNotExist())
 
     // This fails because the album info already exists.
-    assertThatExceptionOfType(ConditionalCheckFailedException::class.java)
-      .isThrownBy {
-        musicTable.albumInfo.save(albumInfo, ifNotExist())
-      }
+    val exception = assertThrows<AmazonServiceException> {
```
Comment (tyiu) [inline tempest/src/test/kotlin/app/cash/tempest/DynamoDbViewTest.kt:79]: Unfortunate consequence of a bug in the DynamoDb Local client where it throws a different exception than what would be thrown in a real-world scenario.

File: tempest/src/test/kotlin/app/cash/tempest/LogicalDbTransactionTest.kt [hunk 4]
```diff
@@ -251,9 +253,10 @@ class LogicalDbTransactionTest {
       )
       .build()
 
-    assertFailsWith<TransactionCanceledException> {
+    val exception = assertThrows<AmazonServiceException> {
```
Comment (tyiu) [inline tempest/src/test/kotlin/app/cash/tempest/LogicalDbTransactionTest.kt:256]: Unfortunate consequence of a bug in the DynamoDb Local client where it throws a different exception than what would be thrown in a real-world scenario.

**Review by zhxnlai — APPROVED** [review]

**Pull request merged**